lib/fluent/plugin/out_logdna.rb

```diff
@@ -14,11 +14,16 @@ class LogDNAOutput < Fluent::BufferedOutput
     config_param :app, :string, default: nil
     config_param :file, :string, default: nil
     config_param :ingester_domain, :string, default: 'https://logs.logdna.com'
-    config_param :request_timeout_seconds, :integer, default: 30
+    config_param :request_timeout, :string, default: '30'
 
     def configure(conf)
       super
       @host = conf['hostname']
+
+      timeout_match = @request_timeout.match(/[0-9]*/)[0]
```

Comment:
This regex will allow numbers such as `000023`. Ruby's `to_i` function should handle it, but do we want that to be allowable syntax? Just raising it to see what you think; I can be a bit of a purist.

Also the regex is free floating and gives a very loose structure on the allowed input. Because `ms_match` is also free floating it will accept inputs such as `ms 30` and `he[ms]llo 250` which seems wrong.

Reply:
Fixed with suggestions below